Build failure: networkmanager-sstp

```
src/nm-sstp-pppd-plugin.c:31:10: fatal error: pppd/chap-new.h: No such file or directory
   31 | #include <pppd/chap-new.h>
      |          ^~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [Makefile:1432: src/nm_sstp_pppd_plugin_la-nm-sstp-pppd-plugin.lo] Error 1
make[2]: Leaving directory '/build/NetworkManager-sstp-1.3.1'
make[1]: *** [Makefile:1596: all-recursive] Error 1
make[1]: Leaving directory '/build/NetworkManager-sstp-1.3.1'
make: *** [Makefile:886: all] Error 2
error: builder for '/nix/store/kznx6s5h5yr39h5b017c4p0hfz7zcjvy-NetworkManager-sstp-gnome-1.3.1.drv' failed with exit code 2;
       last 10 log lines:
       >   CC       src/nm_sstp_pppd_plugin_la-nm-sstp-pppd-plugin.lo
       > src/nm-sstp-pppd-plugin.c:31:10: fatal error: pppd/chap-new.h: No such file or directory
       >    31 | #include <pppd/chap-new.h>
       >       |          ^~~~~~~~~~~~~~~~~
       > compilation terminated.
       > make[2]: *** [Makefile:1432: src/nm_sstp_pppd_plugin_la-nm-sstp-pppd-plugin.lo] Error 1
       > make[2]: Leaving directory '/build/NetworkManager-sstp-1.3.1'
       > make[1]: *** [Makefile:1596: all-recursive] Error 1
       > make[1]: Leaving directory '/build/NetworkManager-sstp-1.3.1'
       > make: *** [Makefile:886: all] Error 2
       For full logs, run 'nix log /nix/store/kznx6s5h5yr39h5b017c4p0hfz7zcjvy-NetworkManager-sstp-gnome-1.3.1.drv'.
```

--- a/pkgs/tools/networking/networkmanager/sstp/default.nix
+++ b/pkgs/tools/networking/networkmanager/sstp/default.nix
@@ -1,6 +1,7 @@
 { stdenv
 , lib
-, fetchurl
+, fetchFromGitLab
+, autoreconfHook
 , file
 , glib
 , gnome
@@ -19,15 +20,19 @@
 
 stdenv.mkDerivation rec {
   pname = "NetworkManager-sstp";
-  version = "1.3.1";
+  version = "unstable-2023-03-09";
   name = "${pname}${lib.optionalString withGnome "-gnome"}-${version}";
 
-  src = fetchurl {
-    url = "mirror://gnome/sources/${pname}/${lib.versions.majorMinor version}/${pname}-${version}.tar.xz";
-    sha256 = "fQMSawiaCk/2ZeMHVVcM7PaFEKbP7bUS9Lh+htrGHX0=";
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "GNOME";
+    repo = "network-manager-sstp";
+    rev = "852db07dc7d19c37e398d831410bd94c8659a210";
+    hash = "sha256-DxgcuTza2G5a7F2mBtDaEuynu7F1Ex9pnAESAjyoRq8=";
   };
 
   nativeBuildInputs = [
+    autoreconfHook
     file
     gettext
     pkg-config
@@ -54,6 +59,7 @@ stdenv.mkDerivation rec {
   configureFlags = [
     "--with-gnome=${if withGnome then "yes" else "no"}"
     "--with-gtk4=${if withGnome then "yes" else "no"}"
+    "--with-pppd-plugin-dir=$(out)/lib/pppd/2.5.0"
     "--enable-absolute-paths"
   ];
 

